langlib/lang.regexp/src/main/ballerina/regexp.bal

```diff
@@ -51,8 +51,40 @@ type GroupsAsSpanArrayType SpanAsTupleType[];
 
 type GroupsArrayType GroupsAsSpanArrayType[];
 
+# Splits a string into substrings separated by matches of a regular expression.
+# This finds the the non-overlapping matches of a regular expression and
+# returns a list of substrings of `str` that occur before the first match,
+# between matches, or after the last match.  If there are no matches, then
+# `[str]` will be returned.
+#
+# ```ballerina
+# regexp:RegExp r = re `,`;
```

Comment:
We expect users to refer to RegExp type using `string:RegExp` since `lang.regexp` is not a predeclared import.
If not users will have to add `lang.regexp` import to use as in the sample. Hope this is fine @mohanvive @MaryamZi 
```suggestion
# string:RegExp r = re `,`;
```

Reply:
+1

Reply:
[f7a5e37](https://github.com/ballerina-platform/ballerina-lang/pull/39096/commits/f7a5e379f25a7a1df588acef1d8a2bb37f3a1a22)

Reply:
If we do `string:RegExp r = re ,;` we will have an issue with doing `r.split()`?